docs/changelog-entries.md

```diff
@@ -0,0 +1,334 @@
+# Changelog Entries
+
+This directory contains individual changelog entry files that are compiled into the main `CHANGELOG.md` during the release process.
+
+## Why Individual Files?
+
+Using separate YAML files for each changelog entry **eliminates merge conflicts** on `CHANGELOG.md` when multiple contributors work on the same branch simultaneously.
+
+## Creating a Changelog Entry
+
+### When to Create an Entry
+
+Create a changelog entry for:
+- ✅ New user-facing features or tools
+- ✅ Breaking changes that affect users or API consumers
+- ✅ Important bug fixes
+- ✅ Significant performance improvements
+- ✅ Dependency updates that affect functionality
+
+**Skip** changelog entries for:
+- ❌ Internal refactoring with no user impact
+- ❌ Documentation-only changes (unless major)
+- ❌ Test-only changes
+- ❌ Minor code cleanup or formatting
+
+### Using the Generator Script (Recommended)
+
+The easiest way to create a changelog entry is using the generator script:
+
+```powershell
+# Interactive mode (prompts for all fields)
+# Defaults to Azure.Mcp.Server
+./eng/scripts/New-ChangelogEntry.ps1
+
+# For a different server
+./eng/scripts/New-ChangelogEntry.ps1 -ServerName "Fabric.Mcp.Server"
+
+# With parameters (defaults to Azure.Mcp.Server)
+./eng/scripts/New-ChangelogEntry.ps1 `
+  -Description "Added support for User-Assigned Managed Identity" `
+  -Section "Features Added" `
+  -PR 1033
+
+# For a specific server
+./eng/scripts/New-ChangelogEntry.ps1 `
+  -ServerName "Fabric.Mcp.Server" `
+  -Description "Added new Fabric workspace tools" `
+  -Section "Features Added" `
+  -PR 1234
+
+# With subsection (automatically title-cased)
+./eng/scripts/New-ChangelogEntry.ps1 `
+  -Description "Updated Azure.Core to 1.2.3" `
+  -Section "Other Changes" `
+  -Subsection "dependency updates" `
+  -PR 1234
+
+# Multi-line description with a list
+$description = @"
+Added new AI Foundry tools:
+- foundry_agents_create: Create a new AI Foundry agent
+- foundry_threads_create: Create a new AI Foundry Agent Thread
+- foundry_threads_list: List all AI Foundry Agent Threads
+"@
+
+./eng/scripts/New-ChangelogEntry.ps1 `
+  -Description $description `
+  -Section "Features Added" `
+  -PR 945
+```
+
+**Features:**
+- Works with any server (Azure.Mcp.Server, Fabric.Mcp.Server, etc.) via `-ServerName` parameter
+- Automatic title-casing of subsections
+- Whitespace trimming from descriptions
+- Support for multi-line descriptions with lists
+- Interactive validation
+
+### Manual Creation
+
+If you prefer to create the file manually:
+
+1. **Generate a unique filename** using the current timestamp in milliseconds:
+   ```powershell
+   # PowerShell
+   [DateTimeOffset]::UtcNow.ToUnixTimeMilliseconds()
+   
+   # Bash
+   date +%s%3N
+   ```
+   Example: `1731260400123.yml`
```

Comment:
No this is a pain in the ass to go through later to tell what each is. Make these alias-description.yaml please